### ISSUE TITLE
Object.instance_method(:class).source_location returns something different than MRI

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1531]
Hm, I'm not sure if this a bug or not. I just noticed this behaviour.

MRI (2.5.3)

```ruby
Object.instance_method(:class).source_location
=> nil
```

TruffleRuby (1.0.0-rc10)

```ruby
Object.instance_method(:class).source_location
=> ["(core)", 1]
```

**Comment by nirvdrum** [comment]
Unless you see this as a problem, I'm leaning towards closing it out as not an issue. The difference you're seeing is that MRI doesn't report source location for any core method, as they're implemented in C. TruffleRuby does report core method source locations because many of them are implemented in Ruby. Seeing that source location allows you to dig into the source and see what the runtime is really doing. However, we also have methods implemented in Java and for those we report the "dummy" source location that you see, which at least still tells you where the method is coming from (i.e., "core"). I suppose we could return `nil` in those cases, but I'm not convinced at the moment that that's a better return value.

To illustrate with another example:

TruffleRuby 1.0.0 RC 11:

```
String.instance_method('gsub').source_location
=> ["resource:/truffleruby/core/string.rb", 998]
```

MRI 2.6.0:

```
String.instance_method('gsub').source_location
=> nil
```

So, strictly speaking, you're correct that this a different behavior than what MRI provides. But unless there's code that relies on `source_location` returning `nil` for core methods, I don't think the difference matters and prefer what we're doing by reporting source location data that we have.

Thoughts?
```

**Comment by deepj** [comment]
@nirvdrum Thanks for the explanation. It looks as not a bug (I haven't been sure about it before). At least it does make sense to me according your points. I guess this can be closed.
